src/test/java/com/puppycrawl/tools/checkstyle/PropertyCacheFileTest.java

```diff
@@ -266,18 +265,94 @@ public void testCacheDirectoryDoesNotExistAndShouldBeCreated() throws IOExceptio
     }
 
     @Test
-    public void testPathToCacheContainsOnlyFileName() throws IOException {
+    public void testPersistCreatesDirectoryIfNotExists() throws IOException {
         final Configuration config = new DefaultConfiguration("myName");
-        final String fileName = "temp.cache";
-        final Path filePath = Paths.get(fileName);
-        final PropertyCacheFile cache = new PropertyCacheFile(config, fileName);
+        final String directoryPath = temporaryFolder + File.separator + "newDir";
+        final String filePath = directoryPath + File.separator + "cache.temp";
+        final PropertyCacheFile cache = new PropertyCacheFile(config, filePath);
+
+        cache.persist();
+
+        final File createdDir = new File(directoryPath);
+        assertWithMessage("Directory should be created")
+                .that(createdDir.exists() && createdDir.isDirectory())
+                .isTrue();
+
+        assertWithMessage("Cache file should be created in the new directory")
+                .that(new File(filePath).exists())
+                .isTrue();
+    }
+
+    @Test
+    public void testPersistWithSymbolicLinkToDirectory() throws IOException {
+        final Path tempDirectory = Files.createTempDirectory("tempDir");
+        final Path symLinkDirectory = Files.createTempDirectory("symLinkDir").resolve("symLink");
+        Files.createSymbolicLink(symLinkDirectory, tempDirectory);
+
+        final Configuration config = new DefaultConfiguration("myName");
+        final String cacheFilePath = symLinkDirectory.resolve("cache.temp").toString();
```

Comment:
This is probably the source of spell check failure

Reply:
done